src/hooks/useLocation.ts

```diff
@@ -0,0 +1,46 @@
+import { useCallback, useEffect, useState } from '@redneckz/uni-jsx/lib/hooks';
+import { useAsyncData } from '@redneckz/uni-jsx/lib/hooks/useAsyncData';
+import { fetchJSONUnsafe } from '../utils/fetchJSON';
+
+const REGION_URL = '/api/v1/region';
+
+export interface RegionData {
+  id?: number;
+  code?: string;
+  name?: string;
+}
+
+interface LocationStorageData {
+  manual?: boolean;
+  region: RegionData;
+}
+
+const localStorageLocation = globalThis?.localStorage.getItem('location');
+const currentLocation: LocationStorageData | null = localStorageLocation
+  ? JSON.parse(localStorageLocation)
+  : null;
+
+export function useLocation(
+  defaultLocation: string,
+): [RegionData, (newLocation: RegionData) => void] {
+  const [location, setLocation] = useState<RegionData>({ name: defaultLocation });
```

Comment:
```suggestion
  const [location, setLocation] = useState<RegionData>(currentLocation ?? { name: defaultLocation });
```

Думаю тут стоит запихать значение, которое мы определяли ранее. 

При первом заходе на сайт пользователь увидит `defaultLocation` из справочника, лишь на секунду, затем подтянется результат из эндпоинта. 
При следующем заходе на сайт, значение, что мы сохраняли ранее в localStorage будет сразу отображено пользователю. И он не увидит "подмены" после ответа эндпоинта